### PR TITLE
Add IExtensibleDisposable which allow external injection of disposable objects

### PR DESCRIPTION
**Pull request by dr1rrb**

### CODE REVIEW

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Uno.Disposables
+{
+	/// <summary>
+	/// Represents a disposable object on which some external disposables can be linked to, in order to share it lifetime.
```
Comment (jeromelaban) [inline src/Uno.Core/Disposables/IExtensibleDisposable.cs]: share its lifetime

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	/// <br />
+	/// private readonly CompositeDisposable _subscriptions = new CompositeDisposable();<br />
+	/// public IDisposable LinkDisposable(IDisposable disposable) => _subscriptions.Register(disposable);<br />
+	/// public void Dispose() => _subscriptions.Dispose();
+	/// </remarks>
+	/// </summary>
+	public interface IExtensibleDisposable : IDisposable
+	{
+		/// <summary>
+		/// Currently registered extensions on the registry
```
Comment (jeromelaban) [inline src/Uno.Core/Disposables/IExtensibleDisposable.cs]: There's no exposed concept of registry, maybe remove "on the registry"

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		/// </summary>
+		/// <remarks>
+		/// To get a known service, use Extensions.OfType&lt;TService&gt;()
```
Comment (jeromelaban) [inline src/Uno.Core/Disposables/IExtensibleDisposable.cs]: Service ?

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		/// </remarks>
+		IReadOnlyCollection<object> Extensions { get; }
+
+		/// <summary>
+		/// Register an extension and link its lifetime (disposal) with the lifetime of the extended disposable
```
Comment (jeromelaban) [inline src/Uno.Core/Disposables/IExtensibleDisposable.cs]: _Registers an extension and links_

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		/// </summary>
+		/// <remarks>
+		/// Extension will be disposed with the extended disposable or when it's unregistered.
```
Comment (jeromelaban) [inline src/Uno.Core/Disposables/IExtensibleDisposable.cs]: _The extension_

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs (hunk 3 above)
Comment (carldebilly) [reply src/Uno.Core/Disposables/IExtensibleDisposable.cs]: `<see />`

**Review by carldebilly — APPROVED** [review]

File: src/Uno.Core/Disposables/IExtensibleDisposable.cs (hunk 3 above)
Comment (dr1rrb) [reply src/Uno.Core/Disposables/IExtensibleDisposable.cs]: "TService" because this interface is designed to attach a 'Service' / 'Behavior' to an object.

Comment (dr1rrb) [reply src/Uno.Core/Disposables/IExtensibleDisposable.cs]: Ok I renamed it to TExtension to be more consistent

**Pull request merged**